command/health/cmd/root.go

```diff
@@ -0,0 +1,185 @@
+	rootCmd.PersistentFlags().StringVar(&desiredState, "desired-state", "passing", "The status state which indicates a service is healthy")
+	rootCmd.PersistentFlags().BoolVar(&prependServiceIdentifier, "prepend-service-identifier", true, "Add 'service:' before the services name, which matches the Consul convention.")
+
+	rootCmd.Flags()
+}
+
+// initConfig reads in config file and ENV variables if set.
+func initConfig() {
+	if cfgFile != "" {
+		// Use config file from the flag.
+		viper.SetConfigFile(cfgFile)
+	} else {
+		// Find home directory.
+		home, err := homedir.Dir()
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		// Search config in home directory with name ".health" (without extension).
+		viper.AddConfigPath(home)
+		viper.SetConfigName(".health")
+	}
+
+	viper.AutomaticEnv() // read in environment variables that match
+
+	// If a config file is found, read it in.
+	if err := viper.ReadInConfig(); err == nil {
+		fmt.Println("Using config file:", viper.ConfigFileUsed())
+	}
+}
+
+// checkServiceStatus verifies the service's status matches the expected state.
+func checkServiceStatus(consulAgentChecksURL, serviceName, jsonPathQueryFormat, successfulStatusState string) bool {
+	jsonResponse := getRawStatusFromConsul(consulAgentChecksURL)
+	serviceStatus := extractResultFromResponse(jsonResponse, jsonPathQueryFormat, serviceName)
+
+	if verboseOutput {
+		fmt.Println(fmt.Sprintf(`
+##########################
+Service Status Evaluation
+--------------------------
+Service's Status: %s
+Acceptable Status: %s 
+##########################`, serviceStatus, successfulStatusState))
+	}
+
+	return serviceStatus == successfulStatusState
+}
+
+// getRawStatusFromConsul retrieves the agent statuses from Consul and unmarshals the result.
+func getRawStatusFromConsul(consulAgentCheckURL string) interface{} {
+	r, err := http.Get(consulAgentCheckURL)
+	if err != nil {
+		fmt.Println(err.Error())
+		os.Exit(1)
+	}
+
+	b, err := ioutil.ReadAll(r.Body)
+	defer r.Body.Close()
+
+	if verboseOutput {
+		fmt.Println(fmt.Sprintf(`
+#################################
+Information from Consul Request
+-------------------------------
+GET Request URL: %s
+Response Status Code: %d
+Response Body: '%s'
+##################################`, consulAgentCheckURL, r.StatusCode, string(b)))
+	}
+
+	var jsonResponse interface{}
+	err = json.Unmarshal(b, &jsonResponse)
+	if err != nil {
+		fmt.Println(err.Error())
+		os.Exit(1)
+	}
+
+	if verboseOutput {
+		fmt.Println("Successfully read and unmarshaled response body.")
+	}
+
+	return jsonResponse
+}
+
+// extractResultFromResponse retrieves the service's status from a jsonResponse using the provided JSON path query and
+// service name.
+func extractResultFromResponse(jsonResponse interface{}, jsonPathFormat, serviceName string) string {
+	serviceHealth, err := jsonpath.JsonPathLookup(jsonResponse, fmt.Sprintf(jsonPathFormat, serviceName))
```

Comment:
This check is implemented incorrectly.

This is checking the keys of the returned JSON object.

The original code checks for a "ServiceName" field in the JSON objects and they match on that.

This makes a difference for the "vault" check and causes the new code to not be equivalent.

Reply:
Good catch. I am going to have to rework this. The JSON path library does not support more complex queries such as `$.[?(@.ServiceName == 'edgex-core-data')].Status` which we could use to get the status of `edgex-core-data`.

Reply:
I re-worked this to just simply serialize the contents from Consul and inspect the results as the script used to do. This allowed me to remove the JSON path library dependency.